sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -579,19 +591,134 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
+          serdeParams.put(serdeConstants.ESCAPE_CHAR, crtTbl.getFieldEscape())
+        }
+
+        if (crtTbl != null && crtTbl.getCollItemDelim() != null) {
+          serdeParams.put(serdeConstants.COLLECTION_DELIM, crtTbl.getCollItemDelim())
+        }
+        if (crtTbl != null && crtTbl.getMapKeyDelim() != null) {
+          serdeParams.put(serdeConstants.MAPKEY_DELIM, crtTbl.getMapKeyDelim())
+        }
+        if (crtTbl != null && crtTbl.getLineDelim() != null) {
+          serdeParams.put(serdeConstants.LINE_DELIM, crtTbl.getLineDelim())
+        }
+
+        if (crtTbl != null && crtTbl.getSerdeProps() != null) {
+          val iter = crtTbl.getSerdeProps().entrySet().iterator()
+          while (iter.hasNext()) {
+            val m = iter.next()
+            serdeParams.put(m.getKey(), m.getValue())
+          }
+        }
+
+        if (crtTbl != null && crtTbl.getComment() != null) {
+          tbl.setProperty("comment", crtTbl.getComment())
+        }
+
+        if (crtTbl != null && crtTbl.getLocation() != null) {
+          HiveShim.setLocation(tbl, crtTbl)
+        }
+
+        if (crtTbl != null) {
+          tbl.setStoredAsSubDirectories(crtTbl.isStoredAsSubDirectories())
+          tbl.setInputFormatClass(crtTbl.getInputFormat())
+          tbl.setOutputFormatClass(crtTbl.getOutputFormat())
+        }
+
+        tbl.getTTable().getSd().setInputFormat(tbl.getInputFormatClass().getName())
+        tbl.getTTable().getSd().setOutputFormat(tbl.getOutputFormatClass().getName())
+
+        if (crtTbl != null && crtTbl.isExternal()) {
+          tbl.setProperty("EXTERNAL", "TRUE")
+          tbl.setTableType(TableType.EXTERNAL_TABLE)
+        }
+
+        // TODO add bucket support
+        // TODO support storage handler
         HiveTable(
           specifiedDatabase = db,
           name = tableName,
-          schema = Seq.empty,
-          partitionColumns = Seq.empty,
-          properties = Map.empty,
-          serdeProperties = Map.empty,
-          tableType = ManagedTable,
-          location = None,
-          inputFormat = None,
-          outputFormat = None,
-          serde = None)
+          schema = tbl.getAllCols.map(field =>
+            HiveColumn(field.getName, field.getType, field.getComment)),
+          partitionColumns = tbl.getPartCols.map(part =>
+            HiveColumn(part.getName, part.getType, part.getComment)),
+          properties = tbl.getParameters.toMap,
+          serdeProperties = serdeParams.toMap,
+          tableType = if (crtTbl.isExternal()) ExternalTable else ManagedTable,
+          location = Some(crtTbl.getLocation()),
+          inputFormat = Some(crtTbl.getInputFormat()),
+          outputFormat = Some(crtTbl.getOutputFormat()),
+          serde = Some(crtTbl.getSerName()))
+      }
+
+      // Get the CreateTableDesc from Hive SemanticAnalyzer
+      val hconf = newHiveConf()
+      val sa = new SemanticAnalyzer(hconf) {
```

Comment:
No, the version of Hive that we use for parsing should not be dictated by the version of Hive that we are using to talk to the metastore.  The problem here is the SemanticAnalyzer does more than parsing, its also doing resolution.  Therefore, it cannot not be used by us since it is connected to a fake metastore that is only used for execution, Thus the tables its looking for will not exist.  This way of doing it is also hacky and tightly couples us with Hive unnecessarily.

Instead, we should parse the the AST as I started doing below, converting it into the version neutral intermediate format so that it can be correctly passed to whatever version of the metastore we are talking to.